src/measure_theory/decomposition/lebesgue.lean

```diff
@@ -0,0 +1,496 @@
+    have hξle : ∀ A, measurable_set A → ∫⁻ a in A, ξ a ∂μ ≤ ν A,
+    { intros A hA, rw hξ,
+      simp_rw [supr_apply],
+      rw lintegral_supr (λ n, (supr_mem_measurable_le _ hf₁ n).1) (supr_monotone _),
+      exact supr_le (λ n, (supr_mem_measurable_le _ hf₁ n).2 A hA) },
+  -- since `E` is positive, we have `∫⁻ a in A ∩ E, ε + ξ a ∂μ ≤ ν (A ∩ E)` for all `A`
+    have hε₂ : ∀ A : set α, measurable_set A →
+      ∫⁻ a in A ∩ E, ε + ξ a ∂μ ≤ ν (A ∩ E),
+    { intros A hA,
+      have := subset_le_of_restrict_le_restrict _ _ hE₁ hE₃ (set.inter_subset_right A E),
+      rwa [zero_apply, to_signed_measure_sub_apply (hA.inter hE₁),
+            measure.sub_apply (hA.inter hE₁) hle,
+            ennreal.to_real_sub_of_le _ (ne_of_lt (measure_lt_top _ _)), sub_nonneg,
+            le_sub_iff_add_le, ← ennreal.to_real_add, ennreal.to_real_le_to_real,
+            measure.coe_nnreal_smul, pi.smul_apply, with_density_apply _ (hA.inter hE₁),
+            show ε • μ (A ∩ E) = (ε : ℝ≥0∞) * μ (A ∩ E), by refl,
+            ← set_lintegral_const, ← lintegral_add measurable_const hξm] at this,
+      { rw [ne.def, ennreal.add_eq_top, not_or_distrib],
+        exact ⟨ne_of_lt (measure_lt_top _ _), ne_of_lt (measure_lt_top _ _)⟩ },
+      { exact ne_of_lt (measure_lt_top _ _) },
+      { exact ne_of_lt (measure_lt_top _ _) },
+      { exact ne_of_lt (measure_lt_top _ _) },
+      { rw with_density_apply _ (hA.inter hE₁),
+        exact hξle (A ∩ E) (hA.inter hE₁) },
+      { apply_instance } },
+  -- from this, we can show `ξ + ε * E.indicator` is a function in `measurable_le` with
+  -- integral greater than `ξ`
+    have hξε : ξ + E.indicator (λ _, ε) ∈ measurable_le μ ν,
+    { refine ⟨measurable.add hξm (measurable.indicator measurable_const hE₁), λ A hA, _⟩,
+      have : ∫⁻ a in A, (ξ + E.indicator (λ _, ε)) a ∂μ =
+            ∫⁻ a in A ∩ E, ε + ξ a ∂μ + ∫⁻ a in A ∩ Eᶜ, ξ a ∂μ,
+      { rw [lintegral_add measurable_const hξm, add_assoc,
+            ← lintegral_union (hA.inter hE₁) (hA.inter (hE₁.compl))
+              (disjoint.mono (set.inter_subset_right _ _) (set.inter_subset_right _ _)
+              disjoint_compl_right), set.inter_union_compl],
+        simp_rw [pi.add_apply],
+        rw [lintegral_add hξm (measurable.indicator measurable_const hE₁), add_comm],
+        refine congr_fun (congr_arg has_add.add _) _,
+        rw [set_lintegral_const, lintegral_indicator _ hE₁, set_lintegral_const,
+            measure.restrict_apply hE₁, set.inter_comm] },
+      conv_rhs { rw ← set.inter_union_compl A E },
+      rw [this, measure_union _ (hA.inter hE₁) (hA.inter hE₁.compl)],
+      { exact add_le_add (hε₂ A hA)
+          (hξle (A ∩ Eᶜ) (hA.inter hE₁.compl)) },
+      { exact disjoint.mono (set.inter_subset_right _ _) (set.inter_subset_right _ _)
+          disjoint_compl_right } },
+      have : ∫⁻ a, ξ a + E.indicator (λ _, ε) a ∂μ ≤ Sup (measurable_le_eval μ ν),
+      { exact le_Sup ⟨ξ + E.indicator (λ _, ε), hξε, rfl⟩ },
```

Comment:
when there is a single tactic closing a goal, don't use `{ exact ... }`, but directly `:= ...`.